libnixt/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, meson
+, ninja
+, pkg-config
+, nix
+, gtest
+, boost182
+}:
+
+stdenv.mkDerivation {
+  pname = "nixt";
+  version = "nightly";
+
+  src = ../.;
+
+  outputs = [ "out" "dev" ];
+
+  mesonBuildType = "release";
+
+  preConfigure = ''
+    cd libnixt
+  '';
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    nix
+    gtest
+    boost182
+  ];
+
+  env.CXXFLAGS = "-include ${nix.dev}/include/nix/config.h";
```

Comment:
nix itself does not `#include` used header by itself, thus we must `#include` it for every file used `nix` library, otherwise the struct layouts maybe different for compiled version.